File: C++/Easy/WrongAnswer/sort-array-by-parity-ii_9.cpp
class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& nums) {
        
        vector<int> result;
        int n = nums.size();
        int odd = 1;
        int even = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] % 2 == 0) {
                result.push_back(nums[i]);
                result.push_back(0);
            } else {
                result.push_back(0);
                result.push_back(nums[i]);
            }
        }
        for (int i = 0; i < n; i++) {
            if (i % 2 == 0) {
                while (result[even] % 2 != 0) {
                    even += 2;
                }
                result[i] = result[even];
                even += 2;
            } else {
                while (result[odd] % 2 == 0) {
                    odd += 2;
                }
                result[i] = result[odd];
                odd += 2;
            }
        }
        return result;
    }
};

File: Rust/Easy/Accepted/sort-array-by-parity-ii_9.rs
impl Solution {
    pub fn sort_array_by_parity_ii(nums: Vec<i32>) -> Vec<i32> {
        
        let mut odd = 1;
        let mut even = 0;
        let mut result = nums.clone();
        for item in nums {
            if item % 2 == 0 {
                result[even] = item;
                even += 2;
            } else {
                result[odd] = item;
                odd += 2;
            }
        }
        result
    }
}